clients/unshaded/src/main/java/tachyon/client/file/options/CreateFileOptions.java

```diff
@@ -0,0 +1,195 @@
+import tachyon.annotation.PublicApi;
+import tachyon.client.ClientContext;
+import tachyon.client.TachyonStorageType;
+import tachyon.client.UnderStorageType;
+import tachyon.client.WriteType;
+import tachyon.client.file.policy.FileWriteLocationPolicy;
+import tachyon.thrift.CreateTOptions;
+import tachyon.util.CommonUtils;
+
+@PublicApi
+public final class CreateFileOptions {
+  private boolean mRecursive;
+  private long mBlockSizeBytes;
+  private FileWriteLocationPolicy mLocationPolicy;
+  private TachyonStorageType mTachyonStorageType;
+  private long mTTL;
+  private UnderStorageType mUnderStorageType;
+
+  /**
+   * @return the default {@link OutStreamOptions}
+   */
+  public static CreateFileOptions defaults() {
+    return new CreateFileOptions();
+  }
+
+  /**
+   * Creates a new instance with defaults from the configuration.
+   */
+  private CreateFileOptions() {
+    mRecursive = false;
+    mBlockSizeBytes = ClientContext.getConf().getBytes(Constants.USER_BLOCK_SIZE_BYTES_DEFAULT);
+    try {
+      mLocationPolicy =
+          CommonUtils
+              .createNewClassInstance(
+                  ClientContext.getConf().<FileWriteLocationPolicy>getClass(
+                      Constants.USER_FILE_WRITE_LOCATION_POLICY),
+                  new Class[]{}, new Object[]{});
+    } catch (Exception e) {
+      throw Throwables.propagate(e);
+    }
+    WriteType defaultWriteType =
+        ClientContext.getConf().getEnum(Constants.USER_FILE_WRITE_TYPE_DEFAULT, WriteType.class);
+    mTachyonStorageType = defaultWriteType.getTachyonStorageType();
+    mUnderStorageType = defaultWriteType.getUnderStorageType();
+    mTTL = Constants.NO_TTL;
+  }
+
+  /**
+   * @return the block size
+   */
+  public long getBlockSizeBytes() {
+    return mBlockSizeBytes;
+  }
+
+  /**
+   * @return the hostname
+   */
+  public FileWriteLocationPolicy getLocationPolicy() {
+    return mLocationPolicy;
+  }
+
+  /**
+   * @return the Tachyon storage type
+   */
+  public TachyonStorageType getTachyonStorageType() {
+    return mTachyonStorageType;
+  }
+
+  /**
+   * @return the TTL (time to live) value; it identifies duration (in milliseconds) the created file
+   *         should be kept around before it is automatically deleted
+   */
+  public long getTTL() {
+    return mTTL;
+  }
+
+  /**
+   * @return the under storage type
+   */
+  public UnderStorageType getUnderStorageType() {
+    return mUnderStorageType;
+  }
+
+  /**
+   * @return whether or not the recursive flag is set
+   */
+  public boolean isRecursive() {
+    return mRecursive;
+  }
+
+  /**
+   * @param blockSizeBytes the block size to use
+   * @return the builder
+   */
+  public CreateFileOptions setBlockSizeBytes(long blockSizeBytes) {
+    mBlockSizeBytes = blockSizeBytes;
+    return this;
+  }
+
+  /**
+   * @param locationPolicy the location policy to use
+   * @return the builder
+   */
+  public CreateFileOptions setHostname(FileWriteLocationPolicy locationPolicy) {
```

Comment:
since there is a `getLocationPolicy`, I guess this should be `setLocationPolicy`?